doc/source/user-environment.rst

```diff
@@ -293,6 +296,103 @@ across sessions. To resolve this, take the following steps:
   which will persist across sessions.
 
 
+.. multiple-profiles:
+
+Using multiple profiles to let users select their environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can create configurations for multiple user environments,
+and let users select from them once they log in to your JupyterHub. This
+is done by creating multiple **profiles**, each of which is attached to a set
+of configuration options that override your JupyterHub's default configuration
+(specified in your Helm Chart). This can be used to let users choose among many
+Docker images, to select the hardware on which they want their jobs to run,
+or to configure default interfaces such as Jupyter Lab vs. RStudio.
+
+Each configuration is a set of options for `Kubespawner <https://github.com/jupyterhub/kubespawner>`_,
+which defines how Kubernetes should launch a new user server pod. Any
+configuration options passed to the `profileList` configuration will
+overwrite the defaults in Kubespawner (or any configuration you've
+added elsewhere in your helm chart).
+
+Profiles are stored under ``singluser: profileList:``, and are defined as
+a list of profiles with specific configuration options each. Here's an example:
+
+.. code-block:: yaml
+
+   singleuser:
+     profileList:
+       - display_name: "Name to be displayed to users"
+         description: "Longer description for users."
+         # Configuration unique to this profile
+         kubespawner_override:
+           your_config: "Your value"
+         # Defines the default profile - only use for one profile
+         default: true
+
+The above configuration will show a screen with information about this profile
+displayed when users start a new server.
+
+Here's an example with two profiles that lets users select the environment they
+wish to use.
+
+.. code-block:: yaml
+
+   singleuser:
+     # Defines the default image
+     image:
+       name: jupyter/minimal-notebook
+       tag: 2343e33dec46
+     profileList:
+       - display_name: "Minimal environment"
+         description: "To avoid too much bells and whistles: Python."
+         default: true
+       - display_name: "Datascience environment"
+         description: "If you want the additional bells and whistles: Python, R, and Julia."
+         kubespawner_override:
+           image: jupyter/datascience-notebook:2343e33dec46
+       - display_name: "Spark environment"
+         description: "The Jupyter Stacks spark image!"
+         kubespawner_override:
+           image: jupyter/all-spark-notebook:2343e33dec46
+
+This allows users to select from three profiles, each with their own
+environment (defined by each Docker image in the configuration above). This is
+equivalent to the following configuration using `extraCongig` to directly define
+configuration for KubeSpawner:
+
+.. code-block:: yaml
+
+   singleuser:
+     image:
+       name: jupyter/minimal-notebook
+       tag: 2343e33dec46
+   hub:
+     extraConfig: |
```

Comment:
While it is still supported to write a string directly to `hub.extraConfig`,  tend to do it like this instead now days. I avoid some warnings about conversion etc. that bugged me as the default value of `hub.extraConfig` is a `{}` and overwriting a dict with a string results in a warning if I recall correctly.

```yaml
hub:
  extraConfig:
    myProfiles: |
      ...
```

@minrk / @yuvipanda got input on this? Examplify with dict or string?

Reply:
I'm happy with whatever you all think is the best practice to recommend to people (we could also remove this section if we're confident that the passthrough works now)